cmd/xl-storage-disk-id-check.go

```diff
@@ -286,8 +286,17 @@ func (p *xlStorageDiskIDCheck) NSScanner(ctx context.Context, cache dataUsageCac
 	}
 
 	weSleep := func() bool {
-		// Entire queue is full, so we sleep.
-		return cap(p.health.tokens) == len(p.health.tokens)
+		switch scannerIdleSpeed.Load() {
+		case 0: // full-throttle
+			return false
+		case 1: // sleep as per the scanner config
+			return true
+		default: // it depends if it is SSD or hard disk
+			if p.storage.rotational {
+				return true
+			}
+			return len(p.health.tokens) < cap(p.health.tokens)
```

Comment:
``` suggestion
			return len(p.health.tokens) < cap(p.health.tokens)
```

This means that `1 < 512` will sleep. @vadmeste I think you need to keep this `len(p.health.tokens) == cap(p.health.tokens)` instead - we don't have to sleep for SSDs unless they are overloaded.

Reply:
> This means that 1 < 512 will sleep

yes, this means the drive is doing 511 parallel operations; the disk is very busy

Reply:
Oh I see now I get it. Forgot that about channels.